.github/workflows/go.yml

```
@@ -28,13 +28,13 @@ jobs:
       run: |
         go install github.com/playwright-community/playwright-go/cmd/playwright
         playwright install --with-deps
-        go test -v ./... -coverprofile=coverage.txt -covermode=atomic
+        go test -v ./... -coverprofile=${{ matrix.os }}_coverage.txt -covermode=atomic
 
-    - name: Codecov
-      uses: codecov/codecov-action@v3
+    - name: Upload coverage report
+      uses: actions/upload-artifact@v3
       with:
-        file: coverage.txt
-        flags: unittests
+        name: reports
+        path: ${{ matrix.os }}_coverage.txt
```

use short retention and ensure no coverage file is an error

```suggestion
        name: reports
        path: ${{ matrix.os }}_coverage.txt
        if-no-files-found: error
        retention-days: 1        
```